### PR TITLE
Release scheduling changes

### PR DESCRIPTION
**Pull request by pietroalbini**
This PR makes four major changes to the release scheduling code:

* Adds easier support for local testing, as explained in the comment at the top of the file.
* Drops support for metadata version 2, which is not supported by publish-release anymore.
* Changed the script to gracefully handle unknown metadata version: commits with unknown metadata will be skipped.
* During scheduled releases (not manual ones) the script will now check the latest released commit for each release, and skip it if it's the same as the commit we're trying to publish. Publishing duplicate releases is already catched and blocked by publish-release, but it would result in a build failure. Hopefully this change will make the Actions page be less red.

### CODE REVIEW

File: ferrocene/ci/scripts/calculate-release-job-matrix.py [hunk 1]
```diff
@@ -278,11 +336,23 @@ class PendingRelease:
...
-    event_data: object
+    manual: bool
+    event_data: dict
+
+    s3: object = field(default_factory=lambda: boto3.client("s3"))
+    http: object = field(default_factory=setup_http_client)
+
+    def env(self):
+        if self.manual:
+            return self.envent_data["inputs"]["env"]
```
Comment (tshepang) [inline ferrocene/ci/scripts/calculate-release-job-matrix.py:348]:
```suggestion
            return self.event_data["inputs"]["env"]
```

**Review by tshepang — APPROVED** [review]
bors merge

**Comment by pietroalbini** [comment]
bors retry

Failed due to #363

**Comment by pietroalbini** [comment]
bors cancel

**Review by tshepang — APPROVED** [review]
bors merge

**Pull request merged**